### PR TITLE
LYNX-179: Documentation for order cancellation mutation

### PR DESCRIPTION
**Pull request by loginesta**
## Purpose of this pull request

This pull request (PR) is to add documentation regarding Order cancellation:
- storeConfig query with new fields (order_cancellation_enabled, order_cancellation_reasons)
- cancelOrder mutation

## Affected pages

src/data/navigation/sections/graphql.js
src/pages/graphql/schema/orders/mutations/cancel-order.md
src/pages/graphql/schema/store/queries/store-config.md

## Links to Magento Open Source code

https://github.com/magento-lynx/magento2ce/pull/145/files

### CODE REVIEW

File: src/pages/graphql/schema/orders/mutations/cancel-order.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+title: cancelOrder mutation | Commerce Web APIs
+---
+
+# cancelOrder mutation
+
+The `cancelOrder` mutation allows the owner of the order and admin users to cancel an order, passing an order identifier and a cancellation reason.
+
+Orders can only be cancelled if `order_cancellation_enabled` config is `true` and if they are in status *RECEIVED* or *PENDING*. If any charge has been done to the customer, the order will be refunded.
```
[the remark this answers is not in the document]
Comment (keharper) [reply src/pages/graphql/schema/orders/mutations/cancel-order.md:9]: You should probably add the status when a refund is required.

File: src/pages/graphql/schema/orders/mutations/cancel-order.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+The mutation returns an error if the order cannot be cancelled.
+
+## Syntax
+
+`mutation: {cancelOrder(input: CanceOrderInput!) {CancelOrderOutput}}`
+
+## Example usage
+
+The following example cancels the order with id `99999999` providing a reason for its cancellation.
```
Comment (keharper) [inline src/pages/graphql/schema/orders/mutations/cancel-order.md:19]:
```suggestion
The following example cancels order ID `99999999` and includes the reason for its cancellation.
```

File: src/pages/graphql/schema/store/queries/store-config.md [hunk 3]
```diff
@@ -243,3 +243,49 @@ The following query returns enumeration values that indicate the store's fixed p
   }
 }
 ```
+
+### Query a store's order cancellation configuration
+
+The following query returns store's order cancellation settings, indicating if it is enabled and the available cancellation reasons.
```
Comment (keharper) [inline src/pages/graphql/schema/store/queries/store-config.md:249]:
```suggestion
The following query returns the store's order cancellation settings, indicating if it is enabled and the available cancellation reasons.
```

File: src/pages/graphql/schema/orders/mutations/cancel-order.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+---
+title: cancelOrder mutation | Commerce Web APIs
+---
+
```
Comment (keharper) [inline src/pages/graphql/schema/orders/mutations/cancel-order.md:4]: Add the following lines above the head1 and be sure to include a blank line before the heading.

```
import BetaNote from '/src/_includes/graphql/notes/beta.md'

<BetaNote />
```

File: src/pages/graphql/schema/store/queries/store-config.md [hunk 5]
```diff
@@ -243,3 +243,49 @@ The following query returns enumeration values that indicate the store's fixed p
   }
 }
 ```
+
+### Query a store's order cancellation configuration
+
```
Comment (keharper) [inline src/pages/graphql/schema/store/queries/store-config.md:248]: Add these lines below the heading

```
import BetaNote from '/src/_includes/graphql/notes/beta.md'

<BetaNote />
```

File: src/pages/graphql/schema/orders/mutations/cancel-order.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (loginesta) [reply src/pages/graphql/schema/orders/mutations/cancel-order.md:9]: what do you mean exactly?

[the remark this answers is not in the document]
Comment (keharper) [reply src/pages/graphql/schema/orders/mutations/cancel-order.md]: Mention the the state of the order when the customer is due a refund.

If the customer has been charged for the order, the order is placed in the X state, and the customer will be issued a refund.

Or, if the order is always placed in the CANCELED state,

The status of the order is set to CANCELED. If the customer has been charged for the order, they will be issued a refund.

**Review by keharper — APPROVED** [review]

**Review by keharper — APPROVED** [review]

**Pull request merged**